client/asset/eth/eth.go

```diff
@@ -1451,9 +1452,20 @@ func (w *assetWallet) swapGas(n int, ver uint32) (oneSwap, nSwap uint64, approve
 	// If we've approved the contract to transfer, we can get a live
 	// estimate to double check.
 
+	// The amount we can estimate and ultimately the amount we can use in a
+	// single transaction is limited by the block gas limit. We attempt to
+	// use one fourth of that. Estimate the number of swaps we can safely
+	// estimate here.
+	maxN := n
+	if n > 1 && nSwap > blockGasLimit {
```

Comment:
My instinct says this should be an error since the inputs are unsatisfiable, and we don't return the reduced value of `n` to the caller so they know what the estimate actually describes.

I had guessed the resolution to https://github.com/decred/dcrdex/issues/2123 would be to modify code a little higher up in `(*assetWallet).estimateSwap` (or perhaps `maxOrder`/`preSwap`) so that they become aware of when `n` is too large, and only return a `asset.SwapEstimate` with the largest possible `lots`

Reply:
Is is bad to scale the estimate? In practice all the SwapAdd are almost the same. The code later will split the swaps at the same break points.

Reply:
It's fine I think to scale up the output of `w.estimateInitGas`, I just thought the problem to solve was higher level callers asking for estimates on impossible swaps...  But now I suppose the thinking it they are only impossible when *batched* but that's not to prevent the user from making a 100 lot order (because it will be split into multiple batches)?

Reply:
I didn't think there were any impossible swaps. Will test this more and see if there is something obviously wrong. Until now large orders were not possible because of the error when trying to get an estimate.